application-admintools-api/src/main/java/com/xwiki/admintools/DataProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools;
+
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+
+/**
+ * Gathers specific data and returns it in a specific format, using a template.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Role
+public interface DataProvider
+{
+    /**
+     * Provides the data provider information in a format given by the associated template.
+     *
+     * @return the information formatted by the associated template as a {@link String}.
+     */
+    String provideData();
+
+    /**
+     * Extract the hint of a component.
+     *
+     * @return the component hint
+     */
+    String getIdentifier();
+
+    /**
+     * Provides the info structured in a json.
+     *
+     * @return a {@link Map} with the generated info.
+     * @throws Exception
+     */
+    Map<String, String> provideJson() throws Exception;
```

Comment:
I also use this method for verifying the health checks, to access the results and verify if there were any errors during the data gathering. Because of this, I would leave this method as it is, as I'm currently using it on that branch and the testing is already done?

Reply:
Ok, if it's going to be used in the future, it can be public from the start, since It would be too much to modify the tests back and forth.
In that case, the name could also be improved. WDYT about `getDataAsJSON` or `getJSONData`? In the same idea, `provideData` could also be updated to maybe `getRenderedData`. The problem I have now is that the names are too generic and don't really say enough about the purpose